### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by minwoox**
Dependencies
- gRPC 1.57.1 -> 1.57.2
- Guava 32.0.1-jre -> 32.1.2-jre
- Logback 1.4.8 -> 1.4.11
- Micrometer 1.11.2 -> 1.11.3
- Protobuf 3.23.4 -> 3.24.0
- Tomcat 10.1.11 -> 10.1.12

Build and test
- Errorprone 2.20.0 -> 2.21.1
- gax-grpc 2.31.1 -> 2.32.0
- Gradle 8.2.1 -> 8.3
- JMH 1.36 -> 1.37
- ktlint-gradle-plugin 11.5.0 -> 11.5.1
- micrometer-tracing 1.1.3 -> 1.1.4

### CODE REVIEW

**Comment by minwoox** [comment]
Might include Netty 4.1.97 if it comes out before we release 1.25.

File: dependencies.toml [hunk 1]
```diff
@@ -479,10 +479,7 @@ version.ref = "krotodc"
 [libraries.guava]
 module = "com.google.guava:guava"
 version.ref = "guava"
-exclusions = [
-    "com.google.errorprone:error_prone_annotations",
-    "com.google.j2objc:j2objc-annotations",
```
Comment (minwoox) [inline dependencies.toml:484]: We cannot exclude these dependencies anymore:
[https://github.com/google/guava/commit/4787cf5591646d25918edc7af29b9f7c1ed4034b#diff-af93ada7ef59cd024b38a39e3f0[…]83e4d6ff77595e3e2879831cddR54-R61](https://github.com/google/guava/commit/4787cf5591646d25918edc7af29b9f7c1ed4034b#diff-af93ada7ef59cd024b38a39e3f0763b91b4b1483e4d6ff77595e3e2879831cddR54-R61)

**Review by trustin — APPROVED** [review]
Thanks, @minwoox! :bow:

**Review by ikhoon — APPROVED** [review]
Thanks, @minwoox!

**Review by jrhee17 — APPROVED** [review]
Thanks! 🙇 👍 🙇

**Pull request merged**